### ISSUE TITLE
Check that values in `exclude:` appear just in `matrix:`

### ISSUE THREAD

**Issue by muzimuzhi** [rhysd/actionlint#634]
Currently in the "[Matrix values](https://github.com/rhysd/actionlint/blob/main/docs/checks.md#matrix-values)" check, `actionlint` checks

> values in `exclude:` appear in `matrix:` or `include:`

But according to the `jobs.<job_id>.strategy.matrix.exclude` doc (emphasis added by me)

> All `include` combinations are processed _after_ `exclude`. This allows you to use `include` to add back combinations that were previously excluded.

`exclude:` values should only be checked if they appear in `matrix:`.

An [example](https://rhysd.github.io/actionlint/#eJyNkM0OgjAQhO88xQQ9igd/LryKMVJgFQzQptsmGuTdtUgR8eKp6c43s5uRTQxluQiCq0w5DgBDbNwLaNtwJF+ATW1jbMRVWfcCGy0MXe5vDKiF0eXN/wDJMQ7eVQmXiOOoqkI5ebvCbjKlW1bZnD4hQNQHfcVM1CFo/2OoRSZ5zi+QlM17QwKhCUrLjJgpx0mcDekTEn9CMrq85b8d402boSVS7K2RazPGYwJTVsg+LFy27dDhWjK6LpxTfeoUew0c9wRvlHge) to play with:

```yaml
on: push

jobs:
  test:
    runs-on: ubuntu-slim
    strategy:
      matrix:
        os: [ ubuntu-latest ]
        php: [ 3, 4 ]
        exclude:
          - os: ubuntu-latest
            php: 5
          - os: macos-latest
        # `include:` are processed _after_ `exclude:`
        include:
          - os: macos-latest
            php: 2
    steps:
      - run: |
          echo os: "${{ matrix.os }}"
          echo php: "${{ matrix.php }}"
```

`actionlint` 1.7.11 gives
```
        exclude:
          - os: ubuntu-latest
            php: 5
                 ^
line:12, col:18 | value "5" in "exclude" does not match in matrix "php" combinations. possible values are "3", "4", "2" matrix
```

For the first `exclude:` pair `{os: ubuntu-latest, php: 5}`, "2" added by `include:` should not be listed in the possible values. 

For the second `exclude:` pair `{os: macos-latest}`, `actionlint` should complain because value "macos-latest" of "os" is only added by `include:`, thus unknown when `exclude:` is processed.

(I found this when checking (out) the latest workflow syntax docs for discussing #630.)
